classes/form/studentattendance.php

```diff
@@ -78,8 +78,8 @@ public function definition() {
                 // Check if an existing status is set, and show it.
                 $existingstatusid = $DB->get_field('attendance_log', 'statusid',
                     ['sessionid' => $attforsession->id, 'studentid' => $USER->id]);
-                $existingstatus = $attblock->get_statuses(false)[$existingstatusid];
                 if (!empty($existingstatus)) {
+                    $existingstatus = $attblock->get_statuses(false)[$existingstatusid];
```

Comment:
this doesn't look right - moving it here will mean the above check will always fail?

Reply:
Thanks, I just did that last commit in a hurry. It's fixed now.